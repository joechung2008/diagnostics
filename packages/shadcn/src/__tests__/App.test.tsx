import App from "@/App";
import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import useSWR from "swr";

vi.mock("swr");

const mockUseSWR = vi.mocked(useSWR);

const mockData = {
  buildInfo: { buildVersion: "1.0.0" },
  extensions: {
    foo: {
      extensionName: "foo",
      config: { X: "1" },
      stageDefinition: { build: ["a"] },
    },
  },
  serverInfo: {
    deploymentId: "d",
    extensionSync: { totalSyncAllCount: 1 },
    hostname: "h",
    nodeVersions: "v1",
    serverId: "s",
    uptime: 1,
  },
};

global.fetch = vi.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(mockData),
  })
) as unknown as typeof window.fetch;

// Mock window.matchMedia
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: vi.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addEventListener: vi.fn(),
    removeEventListener: vi.fn(),
    dispatchEvent: vi.fn(),
  })),
});

describe("App", () => {
  it("renders loading state while loading", async () => {
    mockUseSWR.mockReturnValue({
      data: undefined,
      error: undefined,
      isLoading: true,
      mutate: vi.fn(),
      isValidating: false,
    });
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders error state when diagnostics fail to load", async () => {
    mockUseSWR.mockReturnValue({
      data: undefined,
      error: new Error("Test error"),
      isLoading: false,
      mutate: vi.fn(),
      isValidating: false,
    });
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders initial loaded state diagnostics load", async () => {
    mockUseSWR.mockReturnValue({
      data: mockData,
      error: undefined,
      isLoading: false,
      mutate: vi.fn(),
      isValidating: false,
    });
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
