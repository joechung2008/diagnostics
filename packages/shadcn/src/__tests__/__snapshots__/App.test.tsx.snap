// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders error state when diagnostics fail to load 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1 h-screen"
  >
    <div>
      <nav
        aria-label="Main"
        class="group/navigation-menu relative flex max-w-max flex-1 items-center justify-center"
        data-orientation="horizontal"
        data-slot="navigation-menu"
        data-viewport="true"
        dir="ltr"
      >
        <div
          style="position: relative;"
        >
          <ul
            class="group flex flex-1 list-none items-center justify-center gap-1"
            data-orientation="horizontal"
            data-slot="navigation-menu-list"
            dir="ltr"
          >
            <li
              class="relative"
              data-slot="navigation-menu-item"
            >
              <button
                aria-controls="radix-_r_8_-content-radix-_r_9_"
                aria-expanded="false"
                class="group inline-flex h-9 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground disabled:pointer-events-none disabled:opacity-50 data-[state=open]:hover:bg-accent data-[state=open]:text-accent-foreground data-[state=open]:focus:bg-accent data-[state=open]:bg-accent/50 focus-visible:ring-ring/50 outline-none transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 group"
                data-radix-collection-item=""
                data-slot="navigation-menu-trigger"
                data-state="closed"
                id="radix-_r_8_-trigger-radix-_r_9_"
              >
                Public Cloud 
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down relative top-[1px] ml-1 size-3 transition duration-300 group-data-[state=open]:rotate-180"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </li>
            <li
              class="relative rounded-sm p-2 text-sm hover:bg-accent hover:text-accent-foreground cursor-pointer"
              data-slot="navigation-menu-item"
              style="display: none;"
            >
              paasserverless
            </li>
            <li
              class="relative rounded-sm p-2 text-sm hover:bg-accent hover:text-accent-foreground cursor-pointer"
              data-slot="navigation-menu-item"
            >
              websites
            </li>
          </ul>
        </div>
        <div
          class="absolute top-full left-0 isolate z-50 flex justify-center"
        />
      </nav>
    </div>
    <div
      class="flex flex-col gap-2"
      data-orientation="horizontal"
      data-slot="tabs"
      dir="ltr"
    >
      <div
        aria-orientation="horizontal"
        class="bg-muted text-muted-foreground inline-flex h-9 w-fit items-center justify-center rounded-lg p-[3px]"
        data-orientation="horizontal"
        data-slot="tabs-list"
        role="tablist"
        style="outline: none;"
        tabindex="0"
      >
        <button
          aria-controls="extensions-tab"
          aria-selected="true"
          class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-slot="tabs-trigger"
          data-state="active"
          id="radix-_r_b_-trigger-extensions"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Extensions
        </button>
        <button
          aria-controls="build-tab"
          aria-selected="false"
          class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-slot="tabs-trigger"
          data-state="inactive"
          id="radix-_r_b_-trigger-build"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Build Information
        </button>
        <button
          aria-controls="server-tab"
          aria-selected="false"
          class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-slot="tabs-trigger"
          data-state="inactive"
          id="radix-_r_b_-trigger-server"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Server Information
        </button>
      </div>
    </div>
    <div
      class="flex-1 flex items-center justify-center"
      style="display: none;"
    >
      <svg
        aria-label="Loading"
        class="lucide lucide-loader-circle animate-spin size-20"
        fill="none"
        height="24"
        role="status"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 12a9 9 0 1 1-6.219-8.56"
        />
      </svg>
    </div>
    <div
      class="flex-1 flex items-center justify-center"
    >
      <div>
        <div
          class="relative w-full rounded-lg border px-4 py-3 text-sm grid has-[>svg]:grid-cols-[calc(var(--spacing)*4)_1fr] grid-cols-[0_1fr] has-[>svg]:gap-x-3 gap-y-0.5 items-start [&>svg]:size-4 [&>svg]:translate-y-0.5 text-destructive bg-card [&>svg]:text-current *:data-[slot=alert-description]:text-destructive/90"
          data-slot="alert"
          role="alert"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-circle-alert"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="12"
            />
            <line
              x1="12"
              x2="12.01"
              y1="16"
              y2="16"
            />
          </svg>
          <div
            class="col-start-2 line-clamp-1 min-h-4 font-medium tracking-tight"
            data-slot="alert-title"
          >
            Error
          </div>
          <div
            class="text-muted-foreground col-start-2 grid justify-items-start gap-1 text-sm [&_p]:leading-relaxed"
            data-slot="alert-description"
          >
            Test error
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App > renders initial loaded state diagnostics load 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1 h-screen"
  >
    <div>
      <nav
        aria-label="Main"
        class="group/navigation-menu relative flex max-w-max flex-1 items-center justify-center"
        data-orientation="horizontal"
        data-slot="navigation-menu"
        data-viewport="true"
        dir="ltr"
      >
        <div
          style="position: relative;"
        >
          <ul
            class="group flex flex-1 list-none items-center justify-center gap-1"
            data-orientation="horizontal"
            data-slot="navigation-menu-list"
            dir="ltr"
          >
            <li
              class="relative"
              data-slot="navigation-menu-item"
            >
              <button
                aria-controls="radix-_r_g_-content-radix-_r_h_"
                aria-expanded="false"
                class="group inline-flex h-9 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground disabled:pointer-events-none disabled:opacity-50 data-[state=open]:hover:bg-accent data-[state=open]:text-accent-foreground data-[state=open]:focus:bg-accent data-[state=open]:bg-accent/50 focus-visible:ring-ring/50 outline-none transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 group"
                data-radix-collection-item=""
                data-slot="navigation-menu-trigger"
                data-state="closed"
                id="radix-_r_g_-trigger-radix-_r_h_"
              >
                Public Cloud 
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down relative top-[1px] ml-1 size-3 transition duration-300 group-data-[state=open]:rotate-180"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </li>
            <li
              class="relative rounded-sm p-2 text-sm hover:bg-accent hover:text-accent-foreground cursor-pointer"
              data-slot="navigation-menu-item"
              style="display: none;"
            >
              paasserverless
            </li>
            <li
              class="relative rounded-sm p-2 text-sm hover:bg-accent hover:text-accent-foreground cursor-pointer"
              data-slot="navigation-menu-item"
            >
              websites
            </li>
          </ul>
        </div>
        <div
          class="absolute top-full left-0 isolate z-50 flex justify-center"
        />
      </nav>
    </div>
    <div
      class="flex flex-col gap-2"
      data-orientation="horizontal"
      data-slot="tabs"
      dir="ltr"
    >
      <div
        aria-orientation="horizontal"
        class="bg-muted text-muted-foreground inline-flex h-9 w-fit items-center justify-center rounded-lg p-[3px]"
        data-orientation="horizontal"
        data-slot="tabs-list"
        role="tablist"
        style="outline: none;"
        tabindex="0"
      >
        <button
          aria-controls="extensions-tab"
          aria-selected="true"
          class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-slot="tabs-trigger"
          data-state="active"
          id="radix-_r_j_-trigger-extensions"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Extensions
        </button>
        <button
          aria-controls="build-tab"
          aria-selected="false"
          class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-slot="tabs-trigger"
          data-state="inactive"
          id="radix-_r_j_-trigger-build"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Build Information
        </button>
        <button
          aria-controls="server-tab"
          aria-selected="false"
          class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-slot="tabs-trigger"
          data-state="inactive"
          id="radix-_r_j_-trigger-server"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Server Information
        </button>
      </div>
    </div>
    <div
      class="flex-1 flex items-center justify-center"
      style="display: none;"
    >
      <svg
        aria-label="Loading"
        class="lucide lucide-loader-circle animate-spin size-20"
        fill="none"
        height="24"
        role="status"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 12a9 9 0 1 1-6.219-8.56"
        />
      </svg>
    </div>
    <div
      class="flex-1 flex items-center justify-center"
      style="display: none;"
    >
      <div>
        <div
          class="relative w-full rounded-lg border px-4 py-3 text-sm grid has-[>svg]:grid-cols-[calc(var(--spacing)*4)_1fr] grid-cols-[0_1fr] has-[>svg]:gap-x-3 gap-y-0.5 items-start [&>svg]:size-4 [&>svg]:translate-y-0.5 text-destructive bg-card [&>svg]:text-current *:data-[slot=alert-description]:text-destructive/90"
          data-slot="alert"
          role="alert"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-circle-alert"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="12"
            />
            <line
              x1="12"
              x2="12.01"
              y1="16"
              y2="16"
            />
          </svg>
          <div
            class="col-start-2 line-clamp-1 min-h-4 font-medium tracking-tight"
            data-slot="alert-title"
          >
            Error
          </div>
          <div
            class="text-muted-foreground col-start-2 grid justify-items-start gap-1 text-sm [&_p]:leading-relaxed"
            data-slot="alert-description"
          >
            Unknown error
          </div>
        </div>
      </div>
    </div>
    <div
      class="box-border flex-1 overflow-y-auto"
      id="extensions-tab"
    >
      <div
        class="flex flex-row gap-4 h-full"
      >
        <nav
          aria-label="Extensions"
          class="flex flex-col gap-2 max-h-max overflow-y-auto p-2"
        >
          <button
            class="extension-nav-button w-full min-h-[max-content] box-border cursor-pointer text-left px-2 py-1 rounded hover:bg-accent"
            tabindex="0"
            type="button"
          >
            foo
          </button>
        </nav>
      </div>
    </div>
    <div
      class="box-border flex-1 overflow-y-auto"
      id="build-tab"
      style="display: none;"
    >
      <div
        class="relative w-full overflow-x-auto"
        data-slot="table-container"
      >
        <table
          aria-label="Build Info"
          class="w-full caption-bottom text-sm"
          data-slot="table"
        >
          <thead
            class="[&_tr]:border-b"
            data-slot="table-header"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Name
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
            data-slot="table-body"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                Build Version
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                1.0.0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="box-border flex-1 overflow-y-auto"
      id="server-tab"
      style="display: none;"
    >
      <div
        class="relative w-full overflow-x-auto"
        data-slot="table-container"
      >
        <table
          aria-label="Server Info"
          class="w-full caption-bottom text-sm"
          data-slot="table"
        >
          <thead
            class="[&_tr]:border-b"
            data-slot="table-header"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Name
              </th>
              <th
                class="text-foreground h-10 px-2 text-left align-middle font-medium whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-head"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0"
            data-slot="table-body"
          >
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                Hostname
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                h
              </td>
            </tr>
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                Uptime
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                1
              </td>
            </tr>
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                Server ID
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                s
              </td>
            </tr>
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                Deployment ID
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                d
              </td>
            </tr>
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                Node Versions
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                v1
              </td>
            </tr>
            <tr
              class="hover:bg-muted/50 data-[state=selected]:bg-muted border-b transition-colors"
              data-slot="table-row"
            >
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                Extension Sync | Total Sync All Count
              </td>
              <td
                class="p-2 align-middle whitespace-nowrap [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
                data-slot="table-cell"
              >
                1
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App > renders loading state while loading 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-1 h-screen"
  >
    <div>
      <nav
        aria-label="Main"
        class="group/navigation-menu relative flex max-w-max flex-1 items-center justify-center"
        data-orientation="horizontal"
        data-slot="navigation-menu"
        data-viewport="true"
        dir="ltr"
      >
        <div
          style="position: relative;"
        >
          <ul
            class="group flex flex-1 list-none items-center justify-center gap-1"
            data-orientation="horizontal"
            data-slot="navigation-menu-list"
            dir="ltr"
          >
            <li
              class="relative"
              data-slot="navigation-menu-item"
            >
              <button
                aria-controls="radix-_r_0_-content-radix-_r_1_"
                aria-expanded="false"
                class="group inline-flex h-9 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground disabled:pointer-events-none disabled:opacity-50 data-[state=open]:hover:bg-accent data-[state=open]:text-accent-foreground data-[state=open]:focus:bg-accent data-[state=open]:bg-accent/50 focus-visible:ring-ring/50 outline-none transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 group"
                data-radix-collection-item=""
                data-slot="navigation-menu-trigger"
                data-state="closed"
                id="radix-_r_0_-trigger-radix-_r_1_"
              >
                Public Cloud 
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down relative top-[1px] ml-1 size-3 transition duration-300 group-data-[state=open]:rotate-180"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </li>
            <li
              class="relative rounded-sm p-2 text-sm hover:bg-accent hover:text-accent-foreground cursor-pointer"
              data-slot="navigation-menu-item"
              style="display: none;"
            >
              paasserverless
            </li>
            <li
              class="relative rounded-sm p-2 text-sm hover:bg-accent hover:text-accent-foreground cursor-pointer"
              data-slot="navigation-menu-item"
            >
              websites
            </li>
          </ul>
        </div>
        <div
          class="absolute top-full left-0 isolate z-50 flex justify-center"
        />
      </nav>
    </div>
    <div
      class="flex flex-col gap-2"
      data-orientation="horizontal"
      data-slot="tabs"
      dir="ltr"
    >
      <div
        aria-orientation="horizontal"
        class="bg-muted text-muted-foreground inline-flex h-9 w-fit items-center justify-center rounded-lg p-[3px]"
        data-orientation="horizontal"
        data-slot="tabs-list"
        role="tablist"
        style="outline: none;"
        tabindex="0"
      >
        <button
          aria-controls="extensions-tab"
          aria-selected="true"
          class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-slot="tabs-trigger"
          data-state="active"
          id="radix-_r_3_-trigger-extensions"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Extensions
        </button>
        <button
          aria-controls="build-tab"
          aria-selected="false"
          class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-slot="tabs-trigger"
          data-state="inactive"
          id="radix-_r_3_-trigger-build"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Build Information
        </button>
        <button
          aria-controls="server-tab"
          aria-selected="false"
          class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
          data-orientation="horizontal"
          data-radix-collection-item=""
          data-slot="tabs-trigger"
          data-state="inactive"
          id="radix-_r_3_-trigger-server"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Server Information
        </button>
      </div>
    </div>
    <div
      class="flex-1 flex items-center justify-center"
    >
      <svg
        aria-label="Loading"
        class="lucide lucide-loader-circle animate-spin size-20"
        fill="none"
        height="24"
        role="status"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 12a9 9 0 1 1-6.219-8.56"
        />
      </svg>
    </div>
    <div
      class="flex-1 flex items-center justify-center"
      style="display: none;"
    >
      <div>
        <div
          class="relative w-full rounded-lg border px-4 py-3 text-sm grid has-[>svg]:grid-cols-[calc(var(--spacing)*4)_1fr] grid-cols-[0_1fr] has-[>svg]:gap-x-3 gap-y-0.5 items-start [&>svg]:size-4 [&>svg]:translate-y-0.5 text-destructive bg-card [&>svg]:text-current *:data-[slot=alert-description]:text-destructive/90"
          data-slot="alert"
          role="alert"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-circle-alert"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="8"
              y2="12"
            />
            <line
              x1="12"
              x2="12.01"
              y1="16"
              y2="16"
            />
          </svg>
          <div
            class="col-start-2 line-clamp-1 min-h-4 font-medium tracking-tight"
            data-slot="alert-title"
          >
            Error
          </div>
          <div
            class="text-muted-foreground col-start-2 grid justify-items-start gap-1 text-sm [&_p]:leading-relaxed"
            data-slot="alert-description"
          >
            Unknown error
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
